src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,296 @@
+import * as bootstrap from 'react-bootstrap';
+import CustomInput from '../../../input';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactSelect from 'react-select';
+import SelectWrapper from '../../input/select-wrapper';
+import _ from 'lodash';
+import request from 'superagent';
+
+
+const {Alert, Col, Button, Modal} = bootstrap;
+
+class AddToCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			collectionsAvailable: [],
+			message: {
+				text: null,
+				type: null
+			},
+			selectedCollections: [],
+			showCollectionForm: false
+		};
+
+		this.getCollections = this.getCollections.bind(this);
+		this.handleAddToCollection = this.handleAddToCollection.bind(this);
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleShowCollectionForm = this.handleShowCollectionForm.bind(this);
+		this.handleShowAllCollections = this.handleShowAllCollections.bind(this);
+		this.handleAddToNewCollection = this.handleAddToNewCollection.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	async componentDidMount() {
+		const collections = await this.getCollections();
+		// eslint-disable-next-line react/no-did-mount-set-state
+		this.setState({collectionsAvailable: collections});
+	}
+
+	toggleRow(collectionId) {
+		const oldSelected = this.state.selectedCollections;
+		let newSelected;
+		if (oldSelected.find(selectedId => selectedId === collectionId)) {
+			newSelected = oldSelected.filter(selectedId => selectedId !== collectionId);
+		}
+		else {
+			newSelected = [...oldSelected, collectionId];
+		}
+		this.setState({
+			selectedCollections: newSelected
+		});
+	}
+
+	async getCollections() {
+		// Get all collections of the user (unlikely that a user will have more than 10000 collections
+		const req = await request.get(`/editor/${this.props.userId}/collections/collections?type=${this.props.entityType}&size=10000`);
```

Comment:
It might be a good idea to wrap this in a try/catch block for safety, and display a generic user-friendly error message if there is a failure, something like "Sorry, we could not fetch your collections ({error})"